docs: change specification integration

aepfli opened the pull request.

beeme1mr reviewed (commented):
This may be a good time to introduce the [Docusaurus redirects plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects). It could be configured to redirect calls with a path that starts with `/docs/specification/` to just `/specification/`. I see that you've updated it in this repo but there are likely links to the spec in external blogs and repos.

aepfli commented on docusaurus.config.js:
```diff
@@ -152,13 +151,16 @@ const config = {
+        path: 'external-content/specification/specification',
+        routeBasePath: 'specification',
+        editUrl: (params) => {
+          return `https://github.com/open-feature/specification/edit/main/${params.docPath}`;
```
thank you nice catch

aepfli commented on docusaurus.config.js:
thank you done

aepfli commented:
> This may be a good time to introduce the [Docusaurus redirects plugin](https://docusaurus.io/docs/api/plugins/@docusaurus/plugin-client-redirects). It could be configured to redirect calls with a path that starts with `/docs/specification/` to just `/specification/`. I see that you've updated it in this repo but there are likely links to the spec in external blogs and repos.

I totally agree, i tried to include it in the old path, but i always get a page not found - hence that i moved it one level higher.

I now added the redirect plugin to handle issues for old links - but, it is not recommended to use, we should setup server redirects instead. For now i will leave it - as i did not investigate how we can have server redirects. :)

aepfli commented on docs/tutorials/getting-started/node.mdx:
```diff
@@ -284,4 +284,9 @@ It also showcased how feature flags can be updated at runtime, without requiring
 [nodejs-download]: https://nodejs.org/en/download/
 [docker-download]: https://docs.docker.com/engine/install/
-[flag-key]: https://docs.openfeature.dev/docs/specification/glossary#flag-key
+[flag-key]: https://docs.openfeature.dev/specification/glossary#flag-key
+[error-code]: https://docs.openfeature.dev/specification/types#error-code
+[provider-overview]: https://docs.openfeature.dev/docs/reference/concepts/provider/
+[js-providers]: https://docs.openfeature.dev/docs/reference/technologies/server/javascript/
+[flagd-config]: https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md
+
```
```suggestion
[flag-key]: https://docs.openfeature.dev/specification/glossary#flag-key

```

beeme1mr commented on docs/tutorials/getting-started/go.mdx:
```diff
@@ -242,4 +242,4 @@ It covered how a provider can be configured to perform the flag evaluation and i
 flagging concepts.
 It also showcased how feature flags can be updated at runtime, without requiring a code change and a redeployment.
 
-[flag-key]: https://docs.openfeature.dev/docs/specification/glossary#flag-key
```
Can this be a relative path?

aepfli commented on blog/02-09-2022-creating-a-provider-for-the-go-sdk.mdx:
i will change all the links

aepfli commented on docusaurus.config.js:
```diff
@@ -152,13 +148,29 @@ const config = {
-        sourcePath: 'specification/specification',
-        destinationPath: 'specification',
+        createRedirects(existingPath) {
+          if (existingPath.includes('/docs/specification')) {
```
we might need to change how the preview works - because the preview is still a development build, and the redirects only work in production.

If you like i can schedule a short meeting for the two of use, to verify this locally with a production build.

aepfli replied:
i added netlify redirects, with those we should be safe for the future too :)

aepfli commented:
issue relates to #290
